### ISSUE TITLE
Compiling for ROCm on AMD fails with new setuptools

### ISSUE THREAD

**Issue by owainkenwayucl** [cupy/cupy#8585]
### Description

I've been slowly fighting my way through the issues compiling CuPy 13.3.0 on an AMD cluster I have access to with ROCm 6.1.1 (it's not my cluster).

There are a bunch of deprecations I need to report elsewhere  from ROCm 6.x but the first problem is that if I install the latest `setuptools`, build fail to detect hipblas properly.

The default `setuptools` on the cluster is 65.5.0 which works, but if I update to the latest from `pip` (74.1.1 at time of writing) compiling fails early on with it unable to find files that are... patently there (e.g. `hipblas.h`) and on my includes:

```
Generating cache key from header files...
Cache key (1610 files matching /dev/shm/okenway/cupy/tmp.ZGQsgqHQ64/cupy/cupy/_core/include/**): aacd7ccf6b32f54c63d4eb742fc1188ea94d1b77

-------- Configuring Module: cuda --------
/home/okenway/tmp/tmpgr7uvb35/a.cpp:3:10: fatal error: hipblas.h: No such file or directory
    3 | #include <hipblas.h>
      |          ^~~~~~~~~~~
compilation terminated.
command '/home/software/gcc/14.1.0/bin/g++' failed with exit code 1

************************************************************
* CuPy Configuration Summary                               *
************************************************************

Build Environment:
  Include directories: ['/opt/rocm/include/hipcub', '/dev/shm/okenway/cupy/tmp.ZGQsgqHQ64/cupy/cupy/_core/include', '/opt/rocm/include', '/opt/rocm/include/hip', '/opt/rocm/include/rocrand', '/opt/rocm/include/hiprand', '/opt/rocm/include/roctracer']
  Library directories: ['/opt/rocm/lib']
  nvcc command       : (not found)
  hipcc command      : ['/opt/rocm/bin/hipcc']

Environment Variables:
  CFLAGS          : -I/opt/rocm/include/hipblas -I/opt/rocm/include/hipsparse -I/opt/rocm/include/hipfft -I/opt/rocm/include/rocsolver -I/opt/rocm/include/rccl
  LDFLAGS         : (none)
  LIBRARY_PATH    : /home/software/python/3.11.7/lib:/cm/shared/apps/slurm/current/lib64/slurm:/cm/shared/apps/slurm/current/lib64
  CUDA_PATH       : (none)
  NVCC            : (none)
  HIPCC           : (none)
  ROCM_HOME       : /opt/rocm

Modules:
  cuda      : No
    -> Include files not found: ['hip/hip_runtime_api.h', 'hip/hiprtc.h', 'hipblas.h', 'hiprand/hiprand.h', 'hipsparse.h', 'hipfft.h', 'roctx.h', 'rocsolver.h']
    -> Check your CFLAGS environment variable.

ERROR: CUDA could not be found on your system.

HINT: You are trying to build CuPy from source, which is NOT recommended for general use.
      Please consider using binary packages instead.

Please refer to the Installation Guide for details:
https://docs.cupy.dev/en/stable/install.html

************************************************************

Traceback (most recent call last):
  File "/dev/shm/okenway/cupy/tmp.ZGQsgqHQ64/cupy/setup.py", line 95, in <module>
    ext_modules = cupy_setup_build.get_ext_modules(True, ctx)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/dev/shm/okenway/cupy/tmp.ZGQsgqHQ64/cupy/install/cupy_builder/cupy_setup_build.py", line 514, in get_ext_modules
    extensions = make_extensions(ctx, compiler, use_cython)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/dev/shm/okenway/cupy/tmp.ZGQsgqHQ64/cupy/install/cupy_builder/cupy_setup_build.py", line 363, in make_extensions
    raise Exception('Your CUDA environment is invalid. '
Exception: Your CUDA environment is invalid. Please check above error log.
```

### To Reproduce

1. Create a new virtualenvironment on a system with ROCm 6.x
2. Install latest setuptools (74.1.1) from pip
3. Set environment variables.

```
export ROCM_HOME=/opt/rocm
export HCC_AMDGPU_TARGET=gfx90a
export CFLAGS="-I/opt/rocm/include/hipblas -I/opt/rocm/include/hipsparse -I/opt/rocm/include/hipfft -I/opt/rocm/include/rocsolver -I/opt/rocm/include/rccl"
```

4. Either `pip install cupy` or clone the source and run `python3 setup.py bdist_wheel`

### Installation

None

### Environment

N/A

**Comment by kmaehashi** [comment]
Thanks for the feedback. It's odd as there's should be no difference between CUDA/ROCm regarding header file discovery... Let me take a look.

**Comment by pkubik** [comment]
Seems like it might be fixed in #8319.
Relevant code at https://github.com/cupy/cupy/pull/8319/files#diff-600a8f156dcd88465883d64c262dd953d2e5566f18071d68496a392800e444f4R158
